test/com/google/enterprise/adaptor/opentext/OpentextAdaptorTest.java

```diff
@@ -3049,6 +3067,520 @@ public void testDoEmail() throws IOException {
+        + "</Output>";
+    String response2 =
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>"
+        + "<Output>"
+        + "<SearchResults>"
+        + "<SearchResult>"
+        + "  <OTLocation><![CDATA[2000 1234 5678 -5678 90123]]></OTLocation>"
+        + "  <OTLocationPath>"
+        + "    <LocationPathString>"
+        + "    Enterprise:Folder 1:Project 1"
+        + "    </LocationPathString>"
+        + "  </OTLocationPath> "
+        + "  <OTName>"
+        + "    URL in Project"
+        + "    <Value lang=\"en\">URL in Project</Value>"
+        + "  </OTName>"
+        + "</SearchResult>"
+        + "</SearchResults>"
+        + "<SearchResultsInformation>"
+        + "  <CurrentStartAt>1</CurrentStartAt>"
+        + "  <NumberResultsThisPage>1</NumberResultsThisPage>"
+        + "</SearchResultsInformation>"
+        + "</Output>";
+
+    HttpServer server = startServer(response1, response2, RESPONSE_NO_RESULTS);
+    try {
+      SoapFactoryMock soapFactory = new SoapFactoryMock();
+      OpentextAdaptor adaptor = new OpentextAdaptor(soapFactory);
+      AdaptorContext context = ProxyAdaptorContext.getInstance();
+      Config config = initConfig(adaptor, context);
+      config.overrideKey("opentext.displayUrl.contentServerUrl",
+          "http://127.0.0.1:" + server.getAddress().getPort() + "/");
+      config.overrideKey("opentext.src", "1234, 56780, 12341");
+      adaptor.init(context);
+
+      DocIdPusherMock docIdPusherMock = new DocIdPusherMock();
+      adaptor.getModifiedDocIds(
+          Proxies.newProxyInstance(DocIdPusher.class, docIdPusherMock));
+      assertEquals(Lists.newArrayList(
+              new DocId("1234/Project+1/Document+in+Project:9012"),
+              new DocId("56780/Document+2:90120"),
+              new DocId("12341/Document+3:56781"),
+              new DocId("1234/Project+1/URL+in+Project:90123")),
+          docIdPusherMock.docIds);
+    } finally {
+      server.stop(0);
+    }
+  }
+
+  @Test
+  public void testGetModifiedDocIdsNoResults()
+      throws IOException, InterruptedException {
+    HttpServer server = startServer(RESPONSE_NO_RESULTS);
+    try {
+      SoapFactoryMock soapFactory = new SoapFactoryMock();
+      OpentextAdaptor adaptor = new OpentextAdaptor(soapFactory);
+      AdaptorContext context = ProxyAdaptorContext.getInstance();
+      Config config = initConfig(adaptor, context);
+      config.overrideKey("opentext.displayUrl.contentServerUrl",
+          "http://127.0.0.1:" + server.getAddress().getPort() + "/");
+      config.overrideKey("opentext.src", "1234, 56780, 12341");
+      adaptor.init(context);
+
+      DocIdPusherMock docIdPusherMock = new DocIdPusherMock();
+      adaptor.getModifiedDocIds(
+          Proxies.newProxyInstance(DocIdPusher.class, docIdPusherMock));
+      assertEquals(Lists.newArrayList(), docIdPusherMock.docIds);
+    } finally {
+      server.stop(0);
+    }
+  }
+
+  @Test
+  public void testGetModifiedDocIdsState()
```

Comment:
s/State/Checkpoint/

We just use checkpoint throughout the project to mean query bookmarks like this.